sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -361,6 +361,18 @@ object KeyGroupedPartitioning {
       partitionValues: Seq[InternalRow]): KeyGroupedPartitioning = {
     KeyGroupedPartitioning(expressions, partitionValues.size, Some(partitionValues))
   }
+
+  def supportsExpressions(expressions: Seq[Expression]): Boolean = {
+    def isSupportedTransform(transform: TransformExpression): Boolean = {
+      transform.children.size == 1 && transform.children.head.isInstanceOf[AttributeReference]
+    }
+
+    expressions.forall {
+      case _: AttributeReference => true
```

Comment:
It can't be any other `NamedExpression`, can it?

Reply:
nested column in catalyst is not `AttributeReference`, but `GetStructField(AttributeReference, ...)`. I think we need to support nested column because we support v2 `NamedReference` before.

Reply:
You are right, missed that.
I switched back to `NamedExpression` like in `V2ExpressionUtils`.